src/Nethermind/Nethermind.Core/Collections/ArrayPoolList.cs

```diff
@@ -121,6 +121,49 @@ void ICollection.CopyTo(Array array, int index)
     }
 
     public int Count { get; private set; } = 0;
+    public void ReduceCount(int count)
+    {
+        GuardDispose();
+        var oldCount = Count;
+        if (count == oldCount) return;
+
+        if (count > oldCount)
+        {
+            ThrowOnlyReduce(count);
+        }
+
+        Count = count;
+        if (count < _capacity / 2)
+        {
+            // Reduced to less than half of the capacity, resize the array.
+            T[] newArray = _arrayPool.Rent(count);
+            _array.AsSpan(0, count).CopyTo(newArray);
+            T[] oldArray = Interlocked.Exchange(ref _array, newArray);
+            _capacity = newArray.Length;
+            _arrayPool.Return(oldArray);
```

Comment:
Is it worth it, to potentially copy over this data to the new buffer? 
I expect this data to be consumed fairly quickly and the whole buffer would return.

IMO probably not worth it at

Reply:
The consumers only want 12 or 16 elements by default; so copy isn't that big, but can be hundreds or thousands of nodes